patches/server/0101-world-util-methods-optimization.patch

```diff
@@ -0,0 +1,172 @@
+         // CraftBukkit start - tree generation
+         if (this.captureTreeGeneration) {
+-            BlockState blockstate = null;
+-            Iterator<BlockState> it = capturedBlockStates.iterator();
+-            while (it.hasNext()) {
+-                BlockState previous = it.next();
+-                if (previous.getX() == blockposition.getX() && previous.getY() == blockposition.getY() && previous.getZ() == blockposition.getZ()) {
+-                    blockstate = previous;
+-                    it.remove();
+-                    break;
+-                }
+-            }
+-            if (blockstate == null) {
+-                blockstate = org.bukkit.craftbukkit.block.CraftBlockState.getBlockState(this, blockposition.getX(), blockposition.getY(), blockposition.getZ(), i);
+-            }
+-            blockstate.setTypeId(CraftMagicNumbers.getId(iblockdata.getBlock()));
+-            blockstate.setRawData((byte) iblockdata.getBlock().toLegacyData(iblockdata));
++            this.setCapturedBlockType(blockposition, iblockdata, i);
++            return true;
++        }
++        // CraftBukkit end
++        int x = blockposition.getX();
++        int y = blockposition.getY();
++        int z = blockposition.getZ();
++        Chunk chunk = this.getChunkIfLoaded(x >> 4, z >> 4);
++        if (chunk == null) {
++            return false;
++        }
++
++        if (!this.isValidLocation(x, y, z)) {
++            return false;
++        }
++
++        if (!this.isClientSide && this.worldData.getType() == WorldType.DEBUG_ALL_BLOCK_STATES) {
++            return false;
++        }
++
++        Block block = iblockdata.getBlock();
++
++        // CraftBukkit start - capture blockstates
++        BlockState blockstate = null;
++        if (this.captureBlockStates) {
++            blockstate = org.bukkit.craftbukkit.block.CraftBlockState.getBlockState(this, x, y, z, i);
+             this.capturedBlockStates.add(blockstate);
++        }
++        IBlockData iblockdata1 = chunk.a(blockposition, iblockdata);
++
++        if (iblockdata1 == null) {
++            // CraftBukkit start - remove blockstate if failed
++            if (this.captureBlockStates) {
++                this.capturedBlockStates.remove(blockstate);
++            }
++            // CraftBukkit end
++            return false;
++        }
++        Block block1 = iblockdata1.getBlock();
++
++        if (block.p() != block1.p() || block.r() != block1.r()) {
++            this.methodProfiler.a("checkLight");
++            this.x(blockposition);
++            this.methodProfiler.b();
++        }
++
++        // CraftBukkit start
++        if (!this.captureBlockStates) { // Don't notify clients or update physics while capturing blockstates
++            // Modularize client and physic updates
++            notifyAndUpdatePhysics(blockposition, chunk, block1, block, i);
++        }
++        // CraftBukkit end
++
++        return true;
++    }
++    // Titanium end
```

Comment:
comment
###
There are some more unnecessary comments in this patch, remove them and add the missing panda ones